nasa/harmony
# Harmony 1129 auto pause

## Jira Issue ID
HARMONY-1129

## Description
This PR adds auto-pausing for jobs having more than PREVIEW_THRESHOLD granules. It also activates the `skipPreview` url parameter. When `skipPreview` is `true` there is no auto-pause. 

## Local Test Steps
* Add `PREVIEW_THRESHOLD=10` to your .env file
* Issue a request that will find more than ten granules
* The status page you get redirected to should have a message like
  "The job is generating a preview before auto-pausing. CMR query identified 176 granules, but the request has been limited to process only the first 101 granules because you requested 101 maxResults."
The status should be "previewing"
* Click on the  status page link and refresh the page a few times and you should see the status switch to "paused" with the usual message and 'resume" link
* Either resume or cancel the job to verify it starts running again or switches to the "canceled" status
* Next issue another request with `maxResults` set to 1 and verify that the response is still synchronous.
* Remove the `PREVIEW_THRESHOLD=10` line (or comment it out) in your .env file or you will have many broken tests next time you run `npm run test`

## PR Acceptance Checklist
